src/Serializing/SExpression/Precise.hs

```diff
@@ -23,30 +22,31 @@ pad n = stringUtf8 (replicate (2 * n) ' ')
 
 
 class ToSExpression t where
-  toSExpression :: t -> Int -> Builder
+  toSExpression :: t ann -> Int -> Builder
 
-instance (ToSExpressionWithStrategy strategy t, strategy ~ ToSExpressionStrategy t) => ToSExpression t where
-  toSExpression = toSExpressionWithStrategy @strategy undefined
+instance (ToSExpressionBy strategy t, strategy ~ ToSExpressionStrategy t) => ToSExpression t where
+  toSExpression = toSExpression' @strategy
 
 
-data Strategy = Generic | Show
+data Strategy = Generic | Custom
 
-type family ToSExpressionStrategy t :: Strategy where
-  ToSExpressionStrategy Text = 'Show
-  ToSExpressionStrategy _    = 'Generic
+type family ToSExpressionStrategy (t :: * -> *) :: Strategy where
+  ToSExpressionStrategy (_ :+: _) = 'Custom
+  ToSExpressionStrategy _         = 'Generic
 
-class ToSExpressionWithStrategy (strategy :: Strategy) t where
-  toSExpressionWithStrategy :: proxy strategy -> t -> Int -> Builder
+class ToSExpressionBy (strategy :: Strategy) t where
+  toSExpression' :: t ann -> Int -> Builder
 
-instance Show t => ToSExpressionWithStrategy 'Show t where
-  toSExpressionWithStrategy _ t _ = stringUtf8 (show t)
+instance (ToSExpression l, ToSExpression r) => ToSExpressionBy 'Custom (l :+: r) where
+  toSExpression' (L1 l) = toSExpression l
+  toSExpression' (R1 r) = toSExpression r
```

Comment:
We specialize this instance so as not to show the constructors for anonymous sums, since they don’t add anything meaningful to the tree.